### PR TITLE
format: plots in HEP detailed record

### PR DESCRIPTION
**Pull request by nikpap**
* Displays plots.

Signed-off-by: Nick Papoutsis <admin@nils.gr>

### CODE REVIEW

**Comment by nikpap** [comment]
This is a first draft of how plots could look like:

capture d ecran 2015-11-19 a 11 59 06

Current issues:

1) Some records have too many plots, these appear like this:

capture d ecran 2015-11-19 a 14 18 43

2) Some of the plots descriptions have a very large text that "overflows" the right part of the screen. We should decide where to cut off.

File: inspire/base/static/js/inspire_base_init.js [hunk 1]
```diff
@@ -25,7 +25,8 @@ require([
   "js/feedback_inspire",
   "js/header/header",
   "js/notification",
-  "js/dropdown-menu/dropdown-menu"
+  "js/dropdown-menu/dropdown-menu",
+  "js/plots"
```
Comment (nikpap) [inline inspire/base/static/js/inspire_base_init.js]: @jmartinm We only need to load plots.js in the detailed record, I have a feeling there's another way to do it, other than here.

File: inspire/base/static/js/plots.js [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+  * granted to it by virtue of its status as an Intergovernmental Organization
+  * or submit itself to any jurisdiction.
+  */
+
+  define([
+    'jquery',
+    ],function($) {
+     
+      // $('#plotsCarousel').carousel({
+      //   interval: 500
```
Comment (nikpap) [inline inspire/base/static/js/plots.js]: @jmartinm We can define a carousel internal, after which the next image is automatically loaded. Do we want such a functionality?

File: inspire/base/templates/format/record/Inspire_HTML_detailed_macros.tpl [hunk 3]
```diff
@@ -328,4 +328,80 @@
...
+    <!-- Slider -->
+    <div class="row">
+      <div class="col-xs-12" id="slider">
+        <!-- Top part of the slider -->
+        <div class="row">
+          <div class="col-sm-8" id="carousel-bounding-box">
+            <div class="carousel slide" id="plotsCarousel">
+              <!-- Carousel items -->
+              <div class="carousel-inner">
+                {% for url in record.get('urls') %}
```
Comment (nikpap) [inline inspire/base/templates/format/record/Inspire_HTML_detailed_macros.tpl]: @jmartinm I'm trying to figure a way to skip the second for loop, do you have any suggestions?

The first loop (just as the macro starts) is used to detect if the record has plots, then this loop is used to display them.

File: inspire/base/templates/format/record/Inspire_HTML_detailed_macros.tpl [hunk 4]
```diff
@@ -328,4 +328,80 @@
...
+                  {% if url.get('url').endswith(".png") or url.get('url').endswith(".jpg") %}
```
Comment (nikpap) [inline inspire/base/templates/format/record/Inspire_HTML_detailed_macros.tpl]: @jmartinm I'm assuming plots are .jpg or .png files. Is that actually always the case?

**Comment by salmele** [comment]
>Current issues:
>1) Some records have too many plots, these appear like this:

Should we just have a number of plots in the lower line as large as the window, and then have scroll arrows also on that line? That's what e.g. AirBnB does on the accommodation pics

>2) Some of the plots descriptions have a very large text that "overflows" the right part of the screen. We should decide where to cut off.

Maybe we just cut when we've filled the vertical space ?

**Comment by nikpap** [comment]
Updates on the current issues:

1) In the current implementation, I've opted to initially display the first 5-6 thumbnails. The user can then click on the "Show more plots" button to display additional plots. Because some of our records have a lot of plots, I have the feeling that it'd be best not to implement a vertical/horizontal scroll in the thumbnails. Scrolling through 75 thumbnails to find the one you need might not be that easy for users of our system. It might be based to gather user feedback on this.

2) I've made the text area on the right side have a fixed height. Any text that exceeds this height will "overflow" and the users will be able to scroll through it.

capture d ecran 2015-11-25 a 16 53 00

**Comment by jmartinm** [comment]
> 1) In the current implementation, I've opted to initially display the first 5-6 thumbnails. The user can then click on the "Show more plots" button to display additional plots. Because some of our records have a lot of plots, I have the feeling that it'd be best not to implement a vertical/horizontal scroll in the thumbnails. Scrolling through 75 thumbnails to find the one you need might not be that easy for users of our system. It might be based to gather user feedback on this.

So when you click `Show more`, where are the plots displayed?

> 2) I've made the text area on the right side have a fixed height. Any text that exceeds this height will "overflow" and the users will be able to scroll through it.

This looks like a good solution to me. Depending on how critical this description is it could also be truncated ending with `...` when it does not fit in the available space. You can leave it like this and we gather feedback next week.

**Comment by nikpap** [comment]
>So when you click Show more, where are the plots displayed?

They are currently displayed like before (see second image in this page). Another option would be to add a scrolling bar to the plots thumbnails, like this: 
capture d ecran 2015-11-25 a 17 08 43

**Comment by jmartinm** [comment]
> They are currently displayed like before (see second image in this page). Another option would be to add a scrolling bar to the plots thumbnails, like this: 

Yes, even from the beginning I would display 2 rows and a vertical scrollbar. I think scrolling horizontally is a bit awkward but vertically should be fine.

**Comment by salmele** [comment]
Vertical scrolling seems neat

File: inspire/base/static/js/plots.js [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+ define(
+  [
+    'jquery',
+    'flight/lib/component',
+    'bootstrap'
+  ],
+  function($, defineComponent) {
+      'use strict';
+
+      var sList = [];
```
Comment (jmartinm) [inline inspire/base/static/js/plots.js]: leftover it seems.

File: inspire/base/static/js/plots.js [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+
+      return defineComponent(Plots);
+
+      function Plots() {
+
+        // $('#plotsCarousel').carousel({
```
Comment (jmartinm) [inline inspire/base/static/js/plots.js]: leftover?

**Comment by jmartinm** [comment]
Merged 498aa5de98d3af4c161304a969129ac44e7bbbeb

**Pull request closed without merging**